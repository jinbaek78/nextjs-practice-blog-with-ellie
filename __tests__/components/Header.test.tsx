import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';
import mockRouter from 'next-router-mock';
import Header from '@/components/Header';

describe.skip('Header', () => {
  it('renders correctly', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it('navigates to home page on blog title click', async () => {
    render(<Header />, { wrapper: MemoryRouterProvider });
    await fireEvent.click(screen.getAllByRole('link')[0]);
    expect(mockRouter.asPath).toBe(`/`);
  });

  it('navigates to home page on home button click', async () => {
    render(<Header />, { wrapper: MemoryRouterProvider });
    await fireEvent.click(screen.getByText('home'));
    expect(mockRouter.asPath).toBe(`/`);
  });

  it('navigates to about page on about button click', async () => {
    render(<Header />, { wrapper: MemoryRouterProvider });
    await fireEvent.click(screen.getByText('about'));
    expect(mockRouter.asPath).toBe(`/about`);
  });

  it('navigates to posts page on posts button click', async () => {
    render(<Header />, { wrapper: MemoryRouterProvider });
    await fireEvent.click(screen.getByText('posts'));
    expect(mockRouter.asPath).toBe(`/posts`);
  });

  it('navigates to contact page on contact button click', async () => {
    render(<Header />, { wrapper: MemoryRouterProvider });
    await fireEvent.click(screen.getByText('contact'));
    expect(mockRouter.asPath).toBe(`/contact`);
  });
});
