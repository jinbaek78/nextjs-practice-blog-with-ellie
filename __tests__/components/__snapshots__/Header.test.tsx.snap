// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <header
    class="flex justify-between items-center p-4"
  >
    <a
      href="/"
    >
      <h1
        class="text-3xl font-bold"
      >
        Jin's Blog
      </h1>
    </a>
    <nav
      class="flex gap-4"
    >
      <a
        href="/"
      >
        home
      </a>
      <a
        href="/about"
      >
        about
      </a>
      <a
        href="/posts"
      >
        posts
      </a>
      <a
        href="/contact"
      >
        contact
      </a>
    </nav>
  </header>
</div>
`;
